build/parse.y

```diff
@@ -1029,3 +1045,47 @@ func forceMultiLine(start Position, list []Expr, end Position) bool {
 	elemStart, elemEnd := list[0].Span()
 	return start.Line != elemStart.Line || end.Line != elemEnd.Line
 }
+
+// extractTrailingComment extracts a trailing comment from a block statement
+// and returns the comment (or nil)
+func extractTrailingComment(stmt Expr) *CommentBlock {
+	var body *[]Expr
+	switch block := stmt.(type) {
+	case *DefStmt:
+		body = &block.Body
+	case *ForStmt:
+		body = &block.Body
+	case *IfStmt:
+		// find the innermost node in the if-elif-else chain
+		inner := block
+		for {
```

Comment:
opt: I tend to avoid these sorts of conditionless loops w/ break in favor of standalone funcs that use return.  In this case you could use recursion to walk the chain.

Reply:
Done.